src/main/java/org/broadinstitute/hellbender/tools/walkers/rnaseq/OverhangFixingManager.java

```diff
@@ -251,51 +285,95 @@ protected static boolean isRightOverhang(final GenomeLoc readLoc, final GenomeLo
      *
      * @param read                  the read bases
      * @param readStartIndex        where to start on the read
+     * @param readLength            the length of the read according to the reference, used to prevent overclipping
+     *                              softclipped output from SplitNCigarReads)
      * @param reference             the reference bases
      * @param referenceStartIndex   where to start on the reference
      * @param spanToTest            how many bases to test
      * @return true if too many overhanging bases mismatch, false otherwise
      */
     protected boolean overhangingBasesMismatch(final byte[] read,
                                                final int readStartIndex,
+                                               final int readLength,
                                                final byte[] reference,
                                                final int referenceStartIndex,
                                                final int spanToTest) {
         // don't process too small a span, too large a span, or a span that is most of a read
-        if ( spanToTest < 1 || spanToTest > MAX_BASES_IN_OVERHANG || spanToTest > read.length / 2 ) {
+        if ( spanToTest < 1 || spanToTest > maxBasesInOverhang || spanToTest > readLength / 2 ) {
             return false;
         }
 
         int numMismatchesSeen = 0;
         for ( int i = 0; i < spanToTest; i++ ) {
             if ( read[readStartIndex + i] != reference[referenceStartIndex + i] ) {
-                if ( ++numMismatchesSeen > MAX_MISMATCHES_IN_OVERHANG ) {
+                if ( ++numMismatchesSeen > maxMismatchesInOverhang) {
                     return true;
                 }
             }
         }
 
         // we can still mismatch overall if at least half of the bases mismatch
+
         return numMismatchesSeen >= ((spanToTest+1)/2);
     }
 
     /**
      * Close out the manager stream by clearing the read cache
      */
     public void close() {
+        writeReads(0);
+    }
+
+    /**
+     * Writes read groups off the top of waitingReads until the total number of reads in the set is less than the
+     * target queue size. If outputToFile == false, then the it will instead mark the first item to setMateChanged in
+     * mateChagnedReads
+     */
+    private void writeReads(int targetQueueSize) {
         // write out all of the remaining reads
-        while ( ! waitingReads.isEmpty() ) {
-            writer.addRead(waitingReads.poll().read);
+        while ( getNReadsInQueue() > targetQueueSize ) {
+            List<SplitRead> waitingGroup = waitingReadGroups.poll();
```

Comment:
If we're being really paranoid we might want to make a private class that keeps the count and the read groups and only offers access in a way that they can't get out of synch.  Probably not worth it though since there are very few accesses.